scripts/gallery/team.py

```diff
@@ -0,0 +1,261 @@
+#!/usr/bin/env python3
+
+"""
+This script builds the Team Page Gallery for NumPy.
+It is inspired from the code in pandas/pandas-web project.
+It can be used to build team page for any project that supports
+GitHub https://developer.github.com/v3/repos/statistics/#get-contributors-list-with-additions-deletions-and-commit-counts
+
+It requires as input a team.yml configuration file.
+You can refer to a sample in ./team/numpy folder.
+
+This configuration file should contain:
+```
+team:
+	template_path: <path_to_the_jinja2_templates_directory>
+	base_template: <template_file_all_other_files_will_extend>
+	ignore:
+	- <list_of_files_in_the_source_that_will_not_be_copied_or_processed>
+	github_repo_url: <organization/repo-name>
+	context_preprocessors:
+	- <list_of_functions_that_will_enrich_the_context_parsed_in_this_file>
+	markdown_extensions:
+	- <list_of_markdown_extensions_that_will_be_loaded>
+	maintainers:
+		core: <list_of_core_dev_team_members_github_handle>
+		webteam: <list_of_numpy_webteam_members_github_handle>
+		documentation: <list_of_numpy_documentation_team_members>
+		emeritus: <list_of_emeritus_team_numpy>
+```
+The rest of the items in the file will be added directly to the context.
+
+Run ./team.py --help to see how to run it.
+
+Example:
+
+	* To run the script use:
+		./team.py <Dir Path to yml file location> --ignore-io-errors
+
+	This will pull information about team members listed in yaml file into a local file called team.context. In subsequent runs, unless the team changes, you can reuse this locally stored information to render it using this script and play with layout, css etc. as shown in the next example.
+
+
+	* To run the script using github info stored locally, use: 
+		./team.py <Dir Path to yml file location> --ignore-io-errors --stored-context="team.context"
+"""
+
+import os
+import sys
+import importlib
+import typing
+import markdown
+import shutil
+import time
+import jinja2
+import requests
+import yaml
+import argparse
+
+class Preprocessors:
+	"""
+	Built-in context preprocessors.
+
+	Context preprocessors are functions that receive the context used to
+    render the templates, and enriches it with additional information.
+
+	The original context is obtained by parsing ``team.yml`` and
+	anything else needed just be added with context preprocessors.
+
+	There could be additional context preprocessors such as navbars, blogs
+	for NumPy we only have team context pre-processing for now.  Others may be
+	added later in this script.
+    """
+
+	@staticmethod
+	def team_add_info(context):
+		"""
+		Given the maintainers defined in the yaml file, it fetches
+        the GitHub user information for them.
+        """
+
+		stored_ctx_fname = context["stored_context"]
+		if ( stored_ctx_fname != "" ):
+			with open(stored_ctx_fname) as f:
+				storedctx = yaml.safe_load(f)
+			return storedctx
+
+		context["main"]["maintainers"]["people"] = []
```

Comment:
You can take these 8 lines and put them in a function (say `def get_peopledata(kind):`), and then call that for "core", "webteam", etc. That would get rid of a lot of repeated code and be easier to read.

Reply:
ok

Reply:
Done.